thirdeye-ui/src/app/components/app-bar/app-bar-config-provider/app-bar-config-provider.component.tsx

```diff
@@ -26,21 +26,119 @@ import {
     AppBarConfigProviderPropsContextProps,
 } from "./app-bar-config-provider.interface";
+import Appheader from "../app-header/app-header.component";
+import { useStyle } from "./app-bar.styles";
+import { useGetWorkspaces } from "../../../rest/workspace/workspace-action";
+import { useAuthV1 } from "../../../platform/stores/auth-v1/auth-v1.store";
+import { ActionStatus } from "../../../rest/actions.interfaces";
+import { notifyIfErrors } from "../../../utils/notifications/notifications.util";
+import { useTranslation } from "react-i18next";
+import { useNotificationProviderV1 } from "../../../platform/components";
 
 export const AppBarConfigProvider: FunctionComponent<AppBarConfigProviderProps> =
     ({ children }) => {
+        const { t } = useTranslation();
+        const { notify } = useNotificationProviderV1();
+        const { setWorkspace, workspace } = useAuthV1();
         const [showAppNavBar, setShowAppNavBar] = useState(true);
         const [okToRender, setOkToRender] = useState(false);
+        const [mainViewKey, setMainViewKey] = useState<string | null>(
+            "default"
+        );
+        const [mappedWorkspaces, setMappedWorkspaes] =
+            useState<{ id: string }[]>();
+        const compoenentStyles = useStyle();
+
+        const { workspaces, getWorkspaces, status, errorMessages } =
+            useGetWorkspaces();
+        const isLoading =
+            status === ActionStatus.Initial || status === ActionStatus.Working;
+
+        useEffect(() => {
+            getWorkspaces();
+        }, []);
+
+        useEffect(() => {
+            if (!isEmpty(workspaces)) {
+                /* If the workspace name is not set, we get null as the workspace id,
+                so we map it to a default name to be shown on the UI
+                */
+                setMappedWorkspaes(
+                    workspaces?.map((workspace) => {
+                        if (workspace.id) {
+                            return { id: workspace.id };
+                        } else {
+                            return {
+                                id: t("label.app-header.null-mapped-name"),
+                            };
+                        }
+                    })
+                );
+                setWorkspace(workspaces![0]);
+            }
+        }, [workspaces]);
+
+        const handleWorkspaceChange = (workspace: { id: string }): void => {
+            if (workspace.id === t("label.app-header.null-mapped-name")) {
+                setWorkspace({ id: null });
+            } else {
+                setWorkspace(workspace);
+            }
+        };
+
+        useEffect(() => {
+            /* Using timeout here because we want to re-render once the axios interceptor has been
+            updated with namespace header value, so that API calls will use the updated header.
+            */
+            setTimeout(() => {
+                setMainViewKey(workspace.id);
+            });
```

Comment:
Why do we still need the `setTimeout`? The `useEffect` will run only when the value of the workspace changes.

Reply:
This is to delay the unmounting and mounting of the children till the axios interceptors are updated with the changed workspace. Since AuthProvider is wrapping the entire app, that useEffect runs after the child one, which causes the api calls to use the previous workspace only since interceptor get updated in the useEffect of the component that sits higher up.